src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,159 @@
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+namespace Microsoft.Graph.PowerShell.Authentication.Cmdlets
+{
+    using System;
+    using System.Text;
+    using System.Threading;
+    using System.Net.Http;
+    using System.Collections;
+    using System.Net.Http.Headers;
+    using System.Collections.Generic;
+
+    using System.Management.Automation;
+    using System.Management.Automation.Language;
+    using System.Management.Automation.Runspaces;
+
+    using Newtonsoft.Json;
+    using Microsoft.Graph.PowerShell.Authentication.Helpers;
+
+
+    [Cmdlet(VerbsLifecycle.Invoke, "GraphRequest", DefaultParameterSetName = Constants.UserParameterSet)]
+    [OutputType(typeof(Hashtable))]
+    public class InvokeGraphRequest : PSCmdlet
+    {
+        /// <summary>
+        /// Contains the values of HttpVerbs
+        /// </summary>
+        public enum HttpVerb
+        {
+            GET,
+            POST,
+            PUT,
+            PATCH,
+            DELETE
+        }
+        private readonly HttpClient _client;
+        public InvokeGraphRequest()
+        {
+            this.Method = HttpVerb.GET;
+            var authConfig = GraphSession.Instance.AuthContext;
+            this._client = HttpHelpers.GetGraphHttpClient(authConfig);
+        }
+
+        /// <summary>
+        /// Http Method
+        /// </summary>
+        [Parameter(ParameterSetName = Constants.UserParameterSet, Position = 1, Mandatory = true)]
+        public HttpVerb Method { get; set; }
+
+        /// <summary>
+        /// Uri to call using the Graph HttpClient can be segments such as /beta/me
+        /// or fully qualified url such as https://graph.microsoft.com/beta/me
+        /// </summary>
+        [Parameter(ParameterSetName = Constants.UserParameterSet, Position = 2, Mandatory = true)]
+        public Uri Uri { get; set; }
+
+        /// <summary>
+        /// Optional Http Body
+        /// </summary>
+        [Parameter(ParameterSetName = Constants.UserParameterSet, Position = 3)]
+        public object Body { get; set; }
+
+        protected override void BeginProcessing()
+        {
+            ValidateParameters();
+            base.BeginProcessing();
+        }
+
+        protected override void ProcessRecord()
+        {
+            base.ProcessRecord();
+            var authConfig = GraphSession.Instance.AuthContext;
+            var client = HttpHelpers.GetGraphHttpClient(authConfig);
+            try
+            {
+                var httpMethod = GetHttpMethod();
+                var requestMessage = new HttpRequestMessage(httpMethod, this.Uri)
+                {
+                    Content = new StringContent(null != Body ? JsonConvert.SerializeObject(Body) : @"{}", Encoding.UTF8,
```

Comment:
It is optional to introduce Newtonsoft. The GraphServiceClient abstracts JsonConvert. You can make the following changes to use the GraphServiceClient JSON converters. Eventually it will use System.Text.Json.

- [ ] **Regarding line 80**

`JsonConvert.SerializeObject(Body)`

becomes

`client.HttpProvider.Serializer.SerializeObject(body)`

- [ ] **Regarding line 85**

`JsonConvert.DeserializeObject<Hashtable>(responseString);`

becomes

`client.HttpProvider.Serializer.DeserializeObject<Hashtable>(responseString);`

- [ ] Remove `using Newtonsoft.Json;`

Reply:
Nevermind. You are using the GraphServiceClient's underlying HttpClient.

Reply:
@MIchaelMainer  Since we are using the underlying HttpClient, i'll make this comment as resolved.